controller/client/client.go

```diff
@@ -0,0 +1,97 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/filecoin-project/dealbot/config"
+	"github.com/filecoin-project/dealbot/tasks"
+
+	logging "github.com/ipfs/go-log/v2"
+)
+
+var log = logging.Logger("controller")
+
+// Client is the API client that performs all operations
+// against the dealbot controller.
+type Client struct {
+	client   *http.Client
+	cfg      *config.EnvConfig
+	endpoint string
+}
+
+// New initializes a new API client
+func New(cfg *config.EnvConfig) *Client {
+	endpoint := cfg.Client.Endpoint
+
+	log.Infow("dealbot controller client initialized", "addr", endpoint)
```

Comment:
aim for key/value pairs on these structured logs, iirc

Reply:
I am not sure I understand what you mean? This is already a structured log - i.e. a description + key/value pair.

Reply:
sounds good. i think i missed the 'w' - i've seen others where the description isn't implicitly tagged, so you need to do `"msg", "...", "addr", endpoint`